### PR TITLE
Add disconnection reminder messages to the responder phone (CU-121pbg9)

### PR DESCRIPTION
**Pull request by trisapeace**
- Also added a `locationFactory` function to help with integration test set-up
- Based on this conversation about the same functionality on Buttons side (https://bravely.slack.com/archives/C015WT9JQ1X/p1635276838008900), I changed `sendSingleAlert` to send to both the Responder Phone and to all the Heartbeat Recipients. Eventually the Responder Phone will use the Notification system (to the Alert App), but the Heartbeat Recipients will continue to get text messages.

# Test Plan
- :heavy_check_mark:  Add locations to my local database with `heartbeat_sent_alerts` set to `true`, `false`, and `null`, then run the migration script and see that those with `true` should get the current date, those with `false` or `null` should get `null`
- :heavy_check_mark: Deploy to `dev.sensors.brave.coop`
- :heavy_check_mark: Run smoketests
- :heavy_check_mark: Change the `SUBSEQUENT_VITALS_ALERT_THRESHOLD` value to something reasonable
- :heavy_check_mark: For a server state machine:
   - :heavy_check_mark: Change the `radar_particlecore_id` and `door_particlecode_id` values to simulate only one or the other going down
      -  :heavy_check_mark: See the different disconnection messages depending on which value was changed
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB gets the current time
      - :heavy_check_mark: Wait until `SUBSEQUENT_VITALS_ALERT_THRESHOLD` and see that it gets a disconnection reminder on the heartbeat phones and the responder phone
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB is updated to the current time
      - :heavy_check_mark: Wait `SUBSEQUENT_VITALS_ALERT_THRESHOLD` again and see that it gets a second disconnection reminder on the heartbeat phones and the responder phone
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB is updated to the current time
      - :heavy_check_mark: Change the coreid back and see a reconnection message on the heartbeat phones and the responder phone
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB is updated to `null`
- For a firmware state machine:
   - Change the `radar_particlecore_id` values to simulate it going down
      -  :heavy_check_mark: See the disconnection message
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB gets the current time
      - :heavy_check_mark: Wait until `SUBSEQUENT_VITALS_ALERT_THRESHOLD` and see that it gets a disconnection reminder on the heartbeat phones and the responder phone
         -  :heavy_check_mark: The `sent_vitals_alert_at` value in the DB is updated to the current time
      - :heavy_check_mark: Wait `SUBSEQUENT_VITALS_ALERT_THRESHOLD` again and see that it gets a second disconnection reminder on the heartbeat phones and the responder phone
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB is updated to the current time
      - :heavy_check_mark: Change the coreid back and see a reconnection message on the heartbeat phones and the responder phone
         - :heavy_check_mark: The `sent_vitals_alert_at` value in the DB is updated to `null`
- :heavy_check_mark: Try it with zero, one, and more than one heartbeat phone numbers
- :heavy_check_mark: Try different values of `is_active`

### CODE REVIEW

File: server/index.js
Comment (trisapeace) [inline server/index.js:165]: Write tests for this.

**Comment by trisapeace** [comment]
@schwarrrtz @mariocimet 
To add the `checkHeartbeat` tests, I decided to do that refactor of heartbeat-related code into its own file that I had initially avoided. It made the tests clearer for me to write. But I think this will make it harder to review the changes. So the refactor is in its own commit and the additional tests are in their own commit. Hopefully that makes it easier to look through.

File: server/test/unit/vitalsTest/checkHeartbeatTest.js
Comment (trisapeace) [inline server/test/unit/vitalsTest/checkHeartbeatTest.js:16]: @mariocimet , check this library out, it might be helpful for you too: https://www.npmjs.com/package/rewire

File: server/index.js
Comment (trisapeace) [reply server/index.js]: Done.

File: server/index.js
Comment (trisapeace) [reply server/index.js]: In the end, I did end up doing this.

File: server/test/unit/vitalsTest/checkHeartbeatTest.js
Comment (mariocimet) [reply server/test/unit/vitalsTest/checkHeartbeatTest.js:16]: hah, I had run into this problem before with the state machine, and my solution at the time was to factor out local helpers into another file and import them so they could be stubbed. This feels more elegant

File: server/test/integration/vitalsTest/handleHeartbeatTest.js
Comment (mariocimet) [inline server/test/integration/vitalsTest/handleHeartbeatTest.js:46]: I think this is a forsaken remnant of a copy paste

File: server/db/db.js
Comment (mariocimet) [reply server/db/db.js:315]: this is fine.

**Review by mariocimet — APPROVED** [review]
approved pending the change to the test case description in handleHeartbeatTest.js

File: server/test/integration/vitalsTest/handleHeartbeatTest.js
Comment (trisapeace) [reply server/test/integration/vitalsTest/handleHeartbeatTest.js:46]: Thanks! I'll fix this when I update the migration ID after Dave's pull request is merged.

File: server/db/025-add-sent-vitals-alert-at.sql
Comment (trisapeace) [reply server/db/025-add-sent-vitals-alert-at.sql]: Done.

File: server/test/integration/vitalsTest/handleHeartbeatTest.js
Comment (trisapeace) [reply server/test/integration/vitalsTest/handleHeartbeatTest.js]: Done.

**Comment by trisapeace** [comment]
Task linked: [CU-121pbg9 Send Sensor Disconnection reminders to clients](https://app.clickup.com/t/121pbg9)

**Pull request merged**
